src/main.cpp

```diff
@@ -1,30 +1,143 @@
 #include "../lib/core/types.hpp"
+#include "../lib/logical_solver/presolve.hpp"
 #include "../lib/utils/reader.hpp"
 #include "../lib/utils/primal_reader.hpp"
 #include <vector>
 #include <iostream>
+#include <fstream>
+#include <string>
+#include <chrono>
 
-int main(){
-  utils::Reader reader_;
-  //reader_.readFirstProblem("/home/maxguy/projects/hons/hons-project/problems/feasibility_testcases.txt");
+void testOnTestProblem() {
+  std::string test_problem = "/Users/pepe/hons-project/problems/presolve_test_problem2.txt";
+  utils::Reader reader;
+  // Start filestream to pass to reader.
+  std::fstream problems_filestream;
+  problems_filestream.open(test_problem, std::ios::in);
 
-  reader_.readProblem("/home/maxguy/projects/hons/hons-project/problems/feasibility_testcases.txt", 3);
+  // std::vector<std::vector<int>> problem_matrix;
+  // std::vector<int> upper_bounds;
+  // std::vector<int> lower_bounds;
+  reader.readNextProblem(problems_filestream);
 
-  std::vector<std::vector<core::int_t>> mat = reader_.problem_matrix_;
-  std::vector<core::int_t> max = reader_.upper_bounds_;
-  std::vector<core::int_t> min = reader_.lower_bounds_;
+  logical_solver::Presolve presolve(
+    reader.problem_matrix_, 
+    reader.lower_bounds_, reader.upper_bounds_,
+    reader.num_inequalities_, reader.num_equalities_,
+    false
+  );
+  problems_filestream.close();
 
+  presolve.printLP();
+  presolve.applyPresolve();
+  presolve.applyPostsolve();
+  if (presolve.reduced_to_empty) {
+    printf("Reduced to empty\n");
+  }
+  if (!presolve.infeasible){
+    presolve.printFeasibleSolution();
+  }
+  else {
+    std::cout << "Infeasible" << std::endl;
+  }
+  // presolve.printImpliedBounds();
+}
 
-  for(int i =0; i < mat.size(); ++i){
-    for(int j = 0; j < mat[1].size(); ++j){
-      std::cout << mat[i][j] << std::endl;
-    }
+void testOnSingleProblem(int problem_number) {
+  std::string all_test_problems = "/Users/pepe/hons-project/problems/feasibility_testcases.txt";
+  utils::Reader reader;
+  reader.readProblem(all_test_problems, problem_number);
+
+  logical_solver::Presolve presolve(
+    reader.problem_matrix_, 
+    reader.lower_bounds_, reader.upper_bounds_,
+    reader.num_inequalities_, reader.num_equalities_,
+    true
+  );
+
+  presolve.printLP();
+  presolve.applyPresolve();
+  if (presolve.reduced_to_empty) {
+    printf("Reduced to empty\n");
   }
+  // presolve.applyPostsolve();
+  presolve.printFeasibleSolution();
+  if (presolve.infeasible) {std::cout<<"INFEASIBLE BITCH"<<std::endl;}
```

Comment:
As much as it kills me to say it, I feel like this has to go lol

Reply:
WHOOPS lol